WordPress/Classes/Services/PostRepository.swift

```diff
@@ -63,4 +63,168 @@ final class PostRepository {
         }
     }
 
+    /// Permanently delete the given post from local database and the post's WrodPress site.
+    ///
+    /// - Parameter postID: Object ID of the post
+    func delete<P: AbstractPost>(_ postID: TaggedManagedObjectID<P>) async throws {
+        // Delete the original post instead if presents
+        let original: TaggedManagedObjectID<AbstractPost>? = try await coreDataStack.performQuery { context in
+            let post = try context.existingObject(with: postID)
+            if let original = post.original {
+                return TaggedManagedObjectID(original)
+            }
+            return nil
```

Comment:
Did you consider `guard let`?

```suggestion
            guard let original = post.original else {
                return nil
            }

            return TaggedManagedObjectID(original)
```

Of course the end behavior is the same, but I can't help feeling "safer" with `guard let`.